src/main/java/com/minecraftabnormals/buzzier_bees/core/other/BBEvents.java

```diff
@@ -96,58 +92,30 @@ public static void bottleBug(PlayerInteractEvent.EntityInteractSpecific event) {
 			ItemStack itemstack = event.getPlayer().getHeldItem(event.getHand());
 			Item item = itemstack.getItem();
 
-			Item bottle = null;
-			boolean successful = false;
-
 			Entity target = event.getTarget();
 			EntityType<?> targetType = target.getType();
 			PlayerEntity player = event.getPlayer();
 			Hand hand = event.getHand();
 
-			if (targetType == EntityType.SILVERFISH) {
-				bottle = BBItems.BOTTLE_OF_SILVERFISH.get();
-				successful = true;
-			}
-			if (targetType == EntityType.ENDERMITE) {
-				bottle = BBItems.BOTTLE_OF_ENDERMITE.get();
-				successful = true;
-			}
-			if (targetType == EntityType.BEE) {
-				bottle = BBItems.BOTTLE_OF_BEE.get();
-				successful = true;
-			}
-			ItemStack bottleItem = new ItemStack(bottle);
-
-			if (targetType == EntityType.BEE) {
-				BeeEntity bee = (BeeEntity) target;
-				CompoundNBT tag = bottleItem.getOrCreateTag();
-				tag.putBoolean("HasNectar", bee.hasNectar());
-				tag.putBoolean("HasStung", bee.hasStung());
-				tag.putInt("AngerTime", bee.getAngerTime());
-				if (bee.getAngerTarget() != null) tag.putUniqueId("AngryAt", bee.getAngerTarget());
-				tag.putInt("Age", bee.getGrowingAge());
-				tag.putFloat("Health", bee.getHealth());
-			}
-
-			if (target.hasCustomName()) {
-				ITextComponent name = target.getCustomName();
-				bottleItem = bottleItem.setDisplayName(name);
-			}
-
-			if (successful && ((MobEntity) target).isAlive()) {
-				if (item == Items.GLASS_BOTTLE) {
-					itemstack.shrink(1);
-					event.getWorld().playSound(player, event.getPos(), SoundEvents.ITEM_BOTTLE_FILL_DRAGONBREATH, SoundCategory.NEUTRAL, 1.0F, 1.0F);
-					player.addStat(Stats.ITEM_USED.get(event.getItemStack().getItem()));
-					event.getTarget().remove();
-					if (itemstack.isEmpty()) {
-						player.setHeldItem(hand, bottleItem);
-					} else if (!player.inventory.addItemStackToInventory(bottleItem)) {
-						player.dropItem(bottleItem, false);
+			if (BBCompat.ENTITY_TYPE_TO_BOTTLE_MAP.containsKey(targetType)) {
+				ItemStack bottle = BBCompat.ENTITY_TYPE_TO_BOTTLE_MAP.get(targetType).apply(target);
+
+				if (bottle != null && target.isAlive()) {
+					if (item == Items.GLASS_BOTTLE) {
+						itemstack.shrink(1);
+						event.getWorld().playSound(null, event.getPos(), SoundEvents.ITEM_BOTTLE_FILL_DRAGONBREATH, SoundCategory.NEUTRAL, 1.0F, 1.0F);
+						player.addStat(Stats.ITEM_USED.get(event.getItemStack().getItem()));
```

Comment:
Isn't `event.getItemStack()` the same as the local variable `itemstack`?

Reply:
It seems so, but that's what the original code did. Not exactly sure why though, as I imagine you'd want the glass bottle item and not the new bottled bug item to be counted in the stats.